lxd/main_proxy.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/lxc/lxd/shared"
+)
+
+func cmdProxyDevStart(args *Args) error {
+	err := run(args)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+	return nil
+}
+
+func run(args *Args) error {
+	if len(args.Params) != 5 {
+		return fmt.Errorf("Invalid number of arguments")
+	}
+
+	// Get all our arguments
+	listenPid := args.Params[0]
+	listenAddr := args.Params[1]
+	connectPid := args.Params[2]
+	connectAddr := args.Params[3]
+
+	fd := -1
+	if args.Params[4] != "0" {
+		fd, _ = strconv.Atoi(args.Params[4])
+	}
+
+	// Check where we are in initialization
+	if !shared.PathExists(fmt.Sprintf("/proc/self/fd/%d", fd)) {
+		fmt.Fprintf(os.Stdout, "Listening on %s in %s, forwarding to %s from %s\n", listenAddr, listenPid, connectAddr, connectPid)
```

Comment:
Now that we in the LXD codebase, we can replace all those fmt.Fprintf calls with the proper logging calls.

logger.Debugf would be the appropriate one here.

Reply:
Note that the logger calls are automatically line terminated, so no line breaks should be passed to them.